packages/react-native/src/Authenticator/common/DefaultContainer/DefaultContainer.tsx

```diff
@@ -1,16 +1,56 @@
-import React from 'react';
-import { View } from 'react-native';
+import React, { useMemo } from 'react';
+import { ScrollView, KeyboardAvoidingView } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
 
-import { defaultContainerstyles } from './styles';
+import { useTheme } from '../../../theme';
+import { getThemedStyles } from './styles';
 import { ContainerProps } from './types';
 
+// prevent bounce when `ScrollView` content is less than height of screnn
+const ALWAYS_BOUNCE_VERTICAL = false;
+// prevent keyboard dismiss when press is handled by `ScrollView` children,
+// otherwise dismiss
+const KEYBOARD_SHOULD_PERSIST_TAPS = 'handled';
+// temporary value to prevent top `TextField` from being pushed too high
+// in default use case
+const KEYBOARD_VERTICAL_OFFSET = -200;
+
 export default function DefaultContainer({
+  alwaysBounceVertical = ALWAYS_BOUNCE_VERTICAL,
   children,
+  keyboardAvoidingViewStyle,
+  keyboardShouldPersistTaps = KEYBOARD_SHOULD_PERSIST_TAPS,
+  keyboardVerticalOffset = KEYBOARD_VERTICAL_OFFSET,
+  scrollViewContentContainerStyle,
+  style,
   ...rest
 }: ContainerProps): JSX.Element | null {
+  const theme = useTheme();
+  const { bottom: paddingBottom, top: paddingTop } = useSafeAreaInsets();
+
+  const themedStyle = useMemo(
+    () => getThemedStyles(theme, { paddingBottom, paddingTop }),
+    [paddingBottom, paddingTop, theme]
+  );
+
   return (
-    <View {...rest} style={defaultContainerstyles.container}>
-      {children}
-    </View>
+    <KeyboardAvoidingView
+      behavior="position"
+      keyboardVerticalOffset={keyboardVerticalOffset}
+      style={[themedStyle.keyboardAvoidingView, keyboardAvoidingViewStyle]}
+    >
+      <ScrollView
+        {...rest}
+        alwaysBounceVertical={alwaysBounceVertical}
+        keyboardShouldPersistTaps={keyboardShouldPersistTaps}
+        contentContainerStyle={[
+          themedStyle.scrollViewContentContainer,
+          scrollViewContentContainerStyle,
+        ]}
+        style={[themedStyle.scrollView, style]}
+      >
+        {children}
+      </ScrollView>
+    </KeyboardAvoidingView>
```

Comment:
Have we tested this in landscape mode?

Reply:
Thanks for catching. Had tested an earlier version that was using a `SafeAreaView` on iOS only. Made some improvements for iOS, didn't get in to Android since default React Native apps don't allow orientation updates without native code changes. Am completely open to enabling orientation changes in the RN example and following up in another PR though